File: ueueo-framework/ueueo-eventbus/src/main/java/com/ueueo/eventbus/distributed/OutboxConfigDictionary.cs
using System;
using System.Collections.Generic;

namespace Volo.Abp.EventBus.Distributed;

public class OutboxConfigDictionary : Dictionary<String, OutboxConfig>
{
    public void Configure(Action<OutboxConfig> configAction)
    {
        Configure("Default", configAction);
    }

    public void Configure(String outboxName, Action<OutboxConfig> configAction)
    {
        var outboxConfig = this.GetOrAdd(outboxName, () => new OutboxConfig(outboxName));
        configAction(outboxConfig);
    }
}

File: ueueo-framework/ueueo-eventbus/src/main/java/com/ueueo/eventbus/distributed/OutboxConfigDictionary.java
package com.ueueo.eventbus.distributed;

import java.util.HashMap;
import java.util.function.Consumer;

public class OutboxConfigDictionary extends HashMap<String, OutboxConfig> {
    public void Configure(Consumer<OutboxConfig> configAction) {
        Configure("Default", configAction);
    }

    public void Configure(String outboxName, Consumer<OutboxConfig> configAction) {
        OutboxConfig outboxConfig = computeIfAbsent(outboxName, key -> new OutboxConfig(outboxName));
        configAction.accept(outboxConfig);
    }
}
